website/web/generic_api.py

```diff
@@ -140,12 +143,67 @@ def get(self):
         args = status_parser.parse_args(request)
         task_id = args['task_id']
         seed = args['seed']
+        details = args['details']
         if not seed:
             seed = None
         task = pandora.get_task(task_id=task_id)
         update_user_role(pandora, task, seed)
         assert flask_login.current_user.role.can(Action.read_analysis), 'forbidden'
-        return {'success': True, 'taskId': task.uuid, 'status': task.status.name}
+        to_return = {'success': True, 'taskId': task.uuid, 'status': task.status.name}
+        if details == 1:
+            to_return['workersStatus'] = task.workers_status
+        return to_return
+
+
+worker_parser = api.parser()
+worker_parser.add_argument('task_id', required=True,
+                           location='args',
+                           help="The id of the task you'd like to get the status of")
+worker_parser.add_argument('seed', required=False,
+                           location='args',
+                           help="The seed of the task you'd like to get the status of")
+worker_parser.add_argument('all_workers', type=int, required=False,
+                           location='args',
+                           help="Do you want the details of every workers ? If yes, print 1, else print 0")
+worker_parser.add_argument('worker_name', required=False,
+                           location='args',
+                           help="The name of the worker you want to get the report of")
+worker_parser.add_argument('details', type=int, required=False,
+                           location='args',
+                           help="Do you want the details of the worker status ? If yes, print 1, else print 0")
+
+
+@api.route('/worker_status', methods=['GET'], strict_slashes=False)
+@api.expect(worker_parser)
+class ApiWorkerDetails(Resource):
+
+    @json_answer
+    def get(self):
+        args = worker_parser.parse_args(request)
+        task_id = args['task_id']
+        seed = args['seed']
+        worker_name = args['worker_name']
+        details = args['details']
+        all_workers = args['all_workers']
+        if not seed:
+            seed = None
+        task = pandora.get_task(task_id=task_id)
+        update_user_role(pandora, task, seed)
+        assert flask_login.current_user.role.can(Action.read_analysis), 'forbidden'
+        to_return = {}
+        if all_workers == 1:
+            list_details = []
+            for r in task.reports.values():
+                list_details.append(r.worker_name)
+                list_details.append(r.status.name)
+                list_details.append(r.details)
+            to_return['workers_info'] = list_details
+        else:
+            report = pandora.get_report(task_id, worker_name)
```

Comment:
And here, the return should be the same as above, but with only one entry in the dict.